Fix 3D Tiles Inspector Styling

hpinkos opened the pull request:
Fixes #7028

No need to update `CHANGES.md` since the wrong styling wasn't released

cesium-concierge commented:
Thanks for the pull request @hpinkos!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/tileset-inspector-fix/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

likangning93 commented:
The onscreen area below the inspector no longer seems to detect mouse click events or work for mouseover picking, even with the inspector collapsed. Is this intended?

Also, not introduced in this PR, but sliders don't provide visual feedback in IE. They still work though, so maybe that's not a big deal:

![sliders](https://user-images.githubusercontent.com/5395255/45382551-a2936e80-b5d6-11e8-9271-859bfbd3f069.JPG)

hpinkos commented:
Good catch @likangning93!  We had all this crazy range input styling.  It's better just to use the browser defaults.

This is ready for another look

SunBlack commented on Source/Widgets/Cesium3DTilesInspector/Cesium3DTilesInspector.js:
```diff
@@ -221,27 +225,19 @@ define([
-        toggle.className = 'cesium-cesiumInspector-toggleSwitch';
-        toggle.setAttribute('data-bind', 'text: ' + visibleProp + ' ? "-" : "+", click: ' + toggleProp);
-
+    function makeSection(panel, name, visibleProp, toggleProp, contents) {
```
Synchronize implementation with `CesiumInspector::addSection`. Maybe a base class for both inspectors would be a good idea?

SunBlack commented on Source/Widgets/Cesium3DTilesInspector/Cesium3DTilesInspector.js:
```diff
@@ -52,13 +52,25 @@ define([
         var stylePanelContents = document.createElement('div');
+        stylePanelContents.className = 'cesium-cesiumInspector-sectionContent';
         var optimizationPanelContents = document.createElement('div');
+        optimizationPanelContents.className = 'cesium-cesiumInspector-sectionContent';
```
A lot of code duplication. You can prevent it by using `CesiumInspector::addSection` (remove `makeSection` later).

likangning93 commented on Source/Widgets/Cesium3DTilesInspector/Cesium3DTilesInspector.js:
```diff
@@ -156,22 +170,12 @@ define([
+        makeSection(panel, 'Display', 'displayVisible', 'toggleDisplay', displayPanelContents);
+        makeSection(panel, 'Logging', 'loggingVisible', 'toggleLogging', loggingPanelContents);
+        makeSection(panel, 'Tile Debug Labels', 'tileDebugLabelsVisible', 'toggleTileDebugLabels', tileDebugLabelsPanelContents);
+        makeSection(panel, 'Style', 'styleVisible', 'toggleStyle', stylePanelContents);
```
Looks like the `Update` panel is missing?

likangning93 commented on Source/Widgets/Cesium3DTilesInspector/Cesium3DTilesInspector.js:
```diff
@@ -156,22 +170,12 @@ define([
-        element.appendChild(stylePanel);
-        element.appendChild(optimizationPanel);
+        makeSection(panel, 'Tileset', 'tilesetVisible', 'toggleTileset', tilesetPanelContents);
+        makeSection(panel, 'Display', 'displayVisible', 'toggleDisplay', displayPanelContents);
```
The `Point Cloud Shading` options under `display` used to be hidden when the box was unchecked, is that still doable? If you need it, here's some Sandcastle code for making sure everything is still linking up correctly, but all the parameters seem to work fine as of 75c731b.

```js
var viewer = new Cesium.Viewer('cesiumContainer');
viewer.extend(Cesium.viewerCesium3DTilesInspectorMixin);
var tileset = new Cesium.Cesium3DTileset({ url: Cesium.IonResource.fromAssetId(5713) });
viewer.scene.primitives.add(tileset);
tileset.readyPromise.then(function(){
    viewer.zoomTo(tileset);
});
```

hpinkos commented:
Thanks for the input @SunBlack @likangning93.  That first pass was kind of rushed.

I created an `InspectorShared` static class so both widgets use the same `createCheckbox` and `createSection` functions.  I also fixed the element visibility issues and added back the missing section from the 3d tiles inspector.

Anything else?

SunBlack commented:
Source looks okay for me now 👍 

Just a suggestion: Can't we replace gMinusButton/gPlusButton in CesiumInspector by
`<input type="range" min="1" max=that._numberOfFrustums>`

hpinkos commented:
> Can't we replace gMinusButton/gPlusButton in CesiumInspector

@SunBlack That would be fine with me.  I can't spend any more time on this right now though.  Feel free to open another PR with any more changes =)

likangning93 commented:
Thanks @hpinkos!
